Packages/Ethereum/Sources/SafeDeployments/SafeDeployments.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  File.swift
+//
+//
+//  Created by Dmitry Bespalov on 04.01.22.
+//
+
+import Foundation
+import Solidity
+
+public enum Safe {}
+
+extension Safe {
+    public struct Deployment: Codable {
+        public var defaultAddress: String
+        public var version: String
+        public var abi: Sol.Json.Contract
+        public var networkAddresses: [String: String]
+        public var contractName: String
+        public var released: Bool
+
+        public init(defaultAddress: String, version: String, abi: Sol.Json.Contract, networkAddresses: [String: String], contractName: String, released: Bool) {
+            self.defaultAddress = defaultAddress
+            self.version = version
+            self.abi = abi
+            self.networkAddresses = networkAddresses
+            self.contractName = contractName
+            self.released = released
+        }
+    }
+
+    public enum ContractId: String, Codable {
+        case GnosisSafe = "gnosis_safe"
+        case GnosisSafeL2 = "gnosis_safe_l2"
+        case CreateCall = "create_call"
+        case CreateAndAddModules = "create_and_add_modules"
+        case DefaultCallbackHandler = "default_callback_handler"
+        case CompatibilityFallbackHandler = "compatibility_fallback_handler"
+        case MultiSend = "multi_send"
+        case MultiSendCallOnly = "multi_send_call_only"
+        case ProxyFactory = "proxy_factory"
+        case SignMessageLib = "sign_message_lib"
+        case SimulateTxAccessor = "simulate_tx_accessor"
+    }
+
+    public enum Version: String, Codable {
```

Comment:
I'm not sure if we can do this, but I believe we should count on backend to fetch these constants, because now backend controls which safe version supported or not,

Reply:
We will use backend to decide which actual safe ABI to use, but this is part of the package that generates the ABIs. It doesn't depend on the client gateway.

Reply:
It is based on the Solidity contracts (or rather Json ABI description of contracts of different versions)